corehq/apps/userreports/pillow.py

```diff
@@ -218,6 +219,14 @@ def process_change(self, pillow_instance, change):
         ensure_document_exists(change)
         ensure_matched_revisions(change)
 
+        def exist_in_database(document):
+            query_object = table.get_query_object()
+            if isinstance(query_object, ESAlchemy):
+                records = query_object.es.doc_id(document['_id'])
+            else:
+                records = table.get_query_object().filter_by(doc_id=document['_id'])
+            return records.count > 0
```

Comment:
It would be better to add this function to the `table` object directly and then just call `table.doc_exists(doc_id)`

For the SQL table it would be better to use the `exists` function which is more performant than doing a count: http://stackoverflow.com/a/16003360/632517